### PR TITLE
Simplify tls origination egress gateways

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
1. Use openssl to generate certificates and keys
2. Use client certificate/key
3. Use mutual TLS between the sidecar and the gateway, do not provide both options (with and without mutual TLS)
4. Access my-nginx.mesh-external.svc.cluster.local directly instead of using fake hostname nginx.com. No need for service entries/virtual services
    for nginx.com
5. Use predefined secret names for the istio egress gateway: egressgateway-certs and egressgateway-ca-certs
 6. Use ISTIO_MUTUAL at the egress gateway, for the server's TLS mode

[ ] Configuration Infrastructure
[X ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by adammil2000** [comment]
Can anyone provide a tech review of the proposed changes against our new features in 1.5?

File: content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md [hunk 1]
```diff
@@ -242,15 +184,14 @@ be done by the egress gateway, as opposed to by the sidecar in the previous exam
...
+    $ kubectl exec -it $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -- curl -sL -o /dev/null -D - http://edition.cnn.com/politics
     HTTP/1.1 200 OK
     ...
     content-length: 150793
     ...
     {{< /text >}}
 
-    The output should be the same as in the [TLS Origination for Egress Traffic](/docs/tasks/traffic-management/egress/egress-tls-origination/)
-    example, with TLS origination: without the _301 Moved Permanently_ message.
+    The output should be without the _301 Moved Permanently_ message.
```
Comment (ssuchter) [inline content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md:194]: I think maybe this text would be clearer: "The output should not contain a _301 Moved Permanently_ message."?

Comment (vadimeisenbergibm) [reply content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md]: Fixed in https://github.com/istio/istio.io/pull/6194/commits/ceb7542d4b1d6d2109d1f29b9fd420f6e6159a9d.

**Comment by vadimeisenbergibm** [comment]
@ssuchter Thank you for your review. Let me update the instructions of this PR to Istio 1.5, and then the PR could be merged.

**Comment by vadimeisenbergibm** [comment]
Blocked by https://github.com/istio/istio/issues/23910.

**Comment by frankbu** [comment]
Moved to #7754

**Pull request closed without merging**